spec/hydra/works/models/concerns/generic_file/characterization_spec.rb

```diff
@@ -0,0 +1,119 @@
+require 'spec_helper'
+
+describe Hydra::Works::GenericFile::Characterization do
+  
+
+  # Demonstration class stand in for what downstream applications implement.
+  let(:demoClass)  do
+    class DemoClass < Hydra::Works::GenericFile::Base
+      include Hydra::Works::GenericFile::Characterization
+    end
+  end
+
+  context "using fits.sh via Hydra::FileCharacterization" do
+
+    let(:filename)     { 'sample-file.pdf' }
+    let(:mimetype)     { 'application/pdf' }
+    let(:file_content) { File.open(File.join(fixture_path, filename)) }
+    let(:file)         { Hydra::PCDM::File.new.tap{ |f| f.content = file_content } }
+    let(:type)         { ::RDF::URI("http://pcdm.org/use#OriginalFile") }
+
+    subject { demoClass.new(id:'demo/01/01') }
+
+    before do
+      # Mocking out the Hydra::Works::GenericFile sufficiently to add content, yet not require
+      # a save to fedora
+      allow(file).to receive(:original_name).and_return(filename)
+      allow(file).to receive(:mime_type).and_return(mimetype)
+      # Mock ldp_source.head so AF association believes subject is already saved.
+      allow(subject.ldp_source).to receive(:head).and_return("thefavor")
```

Comment:
This allows for adding the file to generic file without calling save.  It circumvents this by causing the check on if an object is new to return false.  This struck me as a bad plan to be monkeying about with, but I wasn't sure about the extent of the badness.

I might go so far as to mock out a more legitimate `:head` attribute for the `lpd_source` of the GenericFile based on what ActiveFedora would actually stuff in there.